schema/metadata-schema.json

```diff
@@ -0,0 +1,83 @@
+            "title": "Feather Metadata"
+        },
+        "Named": {
+            "type": "object",
+            "additionalProperties": { "type": "string"},
+            "properties": {},
+            "required": [],
+            "title": "Mapped Name"
+        },
+        "ClassMetadata": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "name": { "$ref": "#/definitions/Named" },
+                "owner": { "$ref": "#/definitions/Named" },
+                "security": { "type": "integer", "minimum": 0, "maximum": 65535 },
+                "extends": { "$ref": "#/definitions/Named" },
+                "implements": { "type": "array", "items": { "$ref": "#/definitions/Named" } },
+                "fields": { "type": "array", "items": { "$ref": "#/definitions/FieldMetadata" } },
+                "methods": { "type": "array", "items": { "$ref": "#/definitions/MethodMetadata" } },
+                "inner": { "type": "array", "items": { "$ref": "#/definitions/ClassMetadata" } },
+                "signature": { "$ref": "#/definitions/Named" }
+            },
+            "required": [ "extends", "fields", "implements", "inner", "methods", "name", "owner", "security", "signature" ],
+            "title": "Class Metadata"
+        },
+        "FieldMetadata": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "name": { "$ref": "#/definitions/Named" },
+                "owner": { "$ref": "#/definitions/Named" },
+                "security": { "type": "integer", "minimum": 0, "maximum": 65535 },
+                "descriptor": { "$ref": "#/definitions/Named" },
+                "signature": { "$ref": "#/definitions/Named" }
+            },
+            "required": [ "descriptor", "name", "owner", "security", "signature" ],
+            "title": "Field Metadata"
+        },
+        "MethodMetadata": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "name": { "$ref": "#/definitions/Named" },
+                "owner": { "$ref": "#/definitions/Named" },
+                "security": { "type": "integer", "minimum": 0, "maximum": 65535 },
+                "descriptor": { "$ref": "#/definitions/Named" },
+                "signature": { "$ref": "#/definitions/Named" },
+                "lambda": { "type": "boolean" },
+                "startLine": { "type": "integer", "minimum": 0 },
+                "endLine": { "type": "integer", "minimum": 0 },
+                "bouncingTarget": { "$ref": "#/definitions/BouncingTargetMetadata" },
+                "parent": { "$ref": "#/definitions/MethodMetadata" },
+                "overrides": { "type": "array", "items": { "$ref": "#/definitions/MethodMetadata" } }
+            },
+            "required": [ "descriptor", "name", "owner", "signature" ],
```

Comment:
Not correct there are two types that target methods.
The `MethodMetadata` which includes all the list values here, as well as the `MethodReference` for which this required list would be correct.

As the name suggest `MethodReference` is used in place where a full method is not needed or not even possible due to circular referencing.

`MethodReference`s as such should be used for the following fields in `MethodMetadata`:
- parent

`MethodReference`s should also be used for the following fields in `BouncingTargetMetadata`:
 - target
 - owner

Could you adapt that?

Reply:
Will do, after eating lunch.